### PR TITLE
Make narrowing casts explicit.

### PR DESCRIPTION
**Pull request by ahmedcharles**
#### Summary
Bugfixes "Make narrowing casts explicit."

### CODE REVIEW

**Comment by BrettDong** [comment]
1. C++-style casting is preferred over C-style type casting
2. You need to format your code with AStyle

**Comment by ahmedcharles** [comment]
1. You mean `static_cast`, to be clear?
2. How do I do that?

File: src/condition.cpp [hunk 1]
```diff
@@ -1078,19 +1078,19 @@ std::function<int( const T & )> conditional_t<T>::get_get_int( const JsonObject
         std::string weather_aspect = jo.get_string( "weather" );
         if( weather_aspect == "temperature" ) {
             return []( const T & ) {
-                return get_weather().weather_precise->temperature;
+                return int(get_weather().weather_precise->temperature);
```
Comment (ZhilkinSerg) [inline src/condition.cpp:1081]:
```suggestion
                return static_cast<int>( get_weather().weather_precise->temperature );
```

File: src/condition.cpp [hunk 2]
```diff
@@ -1078,19 +1078,19 @@ std::function<int( const T & )> conditional_t<T>::get_get_int( const JsonObject
...
             };
         } else if( weather_aspect == "windpower" ) {
             return []( const T & ) {
-                return get_weather().weather_precise->windpower;
+                return int(get_weather().weather_precise->windpower);
```
Comment (ZhilkinSerg) [inline src/condition.cpp:1085]:
```suggestion
                return static_cast<int>( get_weather().weather_precise->windpower );
```

File: src/condition.cpp [hunk 3]
```diff
@@ -1078,19 +1078,19 @@ std::function<int( const T & )> conditional_t<T>::get_get_int( const JsonObject
...
             };
         } else if( weather_aspect == "humidity" ) {
             return []( const T & ) {
-                return get_weather().weather_precise->humidity;
+                return int(get_weather().weather_precise->humidity);
```
Comment (ZhilkinSerg) [inline src/condition.cpp:1089]:
```suggestion
                return static_cast<int>( get_weather().weather_precise->humidity );
```

File: src/condition.cpp [hunk 4]
```diff
@@ -1078,19 +1078,19 @@ std::function<int( const T & )> conditional_t<T>::get_get_int( const JsonObject
...
             };
         } else if( weather_aspect == "pressure" ) {
             return []( const T & ) {
-                return get_weather().weather_precise->pressure;
+                return int(get_weather().weather_precise->pressure);
```
Comment (ZhilkinSerg) [inline src/condition.cpp:1093]:
```suggestion
                return static_cast<int>( get_weather().weather_precise->pressure );
```

File: src/condition.cpp [hunk 5]
```diff
@@ -1158,7 +1158,7 @@ std::function<int( const T & )> conditional_t<T>::get_get_int( const JsonObject
                 jo.throw_error( "allies count not supported for NPCs.  In " + jo.str() );
             } else {
                 return []( const T & ) {
-                    return g->allies().size();
+                    return int(g->allies().size());
```
Comment (ZhilkinSerg) [inline src/condition.cpp:1161]:
```suggestion
                    return static_cast<int>( g->allies().size() );
```

File: src/condition.cpp [hunk 6]
```diff
@@ -1209,12 +1209,12 @@ std::function<int( const T & )> conditional_t<T>::get_get_int( const JsonObject
         } else if( checked_value == "power" ) {
             return [is_npc]( const T & d ) {
                 // Energy in milijoule
-                return d.actor( is_npc )->power_cur().value();
+                return int(d.actor( is_npc )->power_cur().value());
```
Comment (ZhilkinSerg) [inline src/condition.cpp:1212]:
```suggestion
                return static_cast<int>( d.actor( is_npc )->power_cur().value() );
```

File: src/condition.cpp [hunk 7]
```diff
@@ -1209,12 +1209,12 @@ std::function<int( const T & )> conditional_t<T>::get_get_int( const JsonObject
...
             };
         } else if( checked_value == "power_max" ) {
             return [is_npc]( const T & d ) {
                 // Energy in milijoule
-                return d.actor( is_npc )->power_max().value();
+                return int(d.actor( is_npc )->power_max().value());
```
Comment (ZhilkinSerg) [inline src/condition.cpp:1217]:
```suggestion
                return static_cast<int>( d.actor( is_npc )->power_max().value() );
```

**Comment by BrettDong** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CODE_STYLE.md

**Pull request merged**